go/webpack/cbor/cbor_reader.go

```diff
@@ -0,0 +1,71 @@
+package cbor
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+)
+
+// A Decoder helps callers iterate through a CBOR buffer, extracting item
+// headers and string bodies as they go.
+type Decoder struct {
+	full []byte
```

Comment:
probably prefer something like cborBytes or cborBuffer but that might just be my verbose C++ history. I leave it up to you.